controllers/storagecluster/webhook.go

```diff
@@ -0,0 +1,142 @@
+package storagecluster
+
+import (
+	"fmt"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+
+	admrv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	OcsMutatingWebhookConfigName = "ocs-operator.ocs.openshift.io"
+	WebhookServiceTargetPort     = 7443
+	webhookServicePort           = 443
+	storageClassWebhookName      = "storageclass.ocs.openshift.io"
+	mutateStorageClassEndpoint   = "/mutate-storageclass"
+	// should be the name from rbac/webhook-service.yaml
+	webhookServiceName = "ocs-operator-webhook-server"
+)
+
+type storageClassWebhook struct{}
+
+var _ resourceManager = &storageClassWebhook{}
+
+// should match the spec at rbac/webhook-service.yaml
+var webhookService = corev1.Service{
+	Spec: corev1.ServiceSpec{
+		Ports: []corev1.ServicePort{
+			{
+				Name:       "ocs-operator-webhook",
+				Port:       webhookServicePort,
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.FromInt32(WebhookServiceTargetPort),
+			},
+		},
+		Selector: map[string]string{
+			"name": "ocs-operator",
+		},
+		Type: corev1.ServiceTypeClusterIP,
+	},
+}
+
+var storageClassMutatingWebhook = admrv1.MutatingWebhook{
```

Comment:
do we need service and webhook created as part of the code not part of yaml files in the CSV? can you please specify the reason and pros and cons?

Reply:
svc:
I want to ensure the annotation even if it's removed by mistake and having the svc in bundle doesn't ensure it

webhook:
I need support for setting MatchConditions which aren't exposed on CSV, need typed structs + enforcement and so not part of bundle